daml-foundations/daml-ghc/damldoc/src/DA/Daml/GHC/Damldoc/Render/Rst.hs

```diff
@@ -150,71 +155,44 @@ fieldTable fds = T.unlines $ -- NB final empty line is essential and intended
                 , "  - Type"
                 , "  - Description" ]
     fieldRows = concat
-       [ [ prefix "* - " $ escapeTr_ fd_name
+       [ [ prefix "* - " $ escapeTr_ (unFieldname fd_name)
          , prefix "  - " $ type2rst fd_type
-         , prefix "  - " $ maybe " " (markdownToRst . T.unwords . T.lines) fd_descr ]
+         , prefix "  - " $ maybe " " (docTextToRst . DocText . T.unwords . T.lines . unDocText) fd_descr ] -- FIXME: this makes no sense
```

Comment:
That construction `unwords . lines` might seem funny... the idea was to ensure the documentation text is a single line. Otherwise we need to  indent it correctly, which is a bit fiddly. 

However, it might be better to do that. As a principle, we should retain the line breaks in doc.text over multiple lines. More important for doc.s on modules and top-level declarations, though.

Reply:
Yeah, makes sense. I was thinking of altering the representation of the output of Render.RST (and Render.Markdown, which has the same problem) to make it easier to indent things correctly. Some kind of "rendered doc" IR that can be converted to Rst or Markdown easily. Because right now there's a lot of duplication between the output modes of damldoc.